File: examples/node/quickstart.mjs
/**
 * Minimal Node.js quickstart for connecting to the TweekIT MCP server.
 *
 * Usage:
 *   node examples/node/quickstart.mjs --file path/to/input.pdf --outfmt txt
 *
 * Requires: npm install @modelcontextprotocol/sdk
 */

import fs from "node:fs/promises";
import path from "node:path";
import process from "node:process";
import { argv } from "node:process";

import { Client } from "@modelcontextprotocol/sdk/client/index.js";

const DEFAULT_SERVER = process.env.TWEEKIT_MCP_SERVER ?? "https://mcp.tweekit.com/mcp";
const API_KEY = process.env.TWEAKIT_API_KEY;
const API_SECRET = process.env.TWEAKIT_API_SECRET;

function parseArgs() {
  const args = new URLSearchParams();
  for (let i = 2; i < argv.length; i += 2) {
    const key = argv[i];
    const value = argv[i + 1];
    if (!key?.startsWith("--") || value === undefined) {
      throw new Error("Arguments must be provided as --flag value pairs.");
    }
    args.set(key.slice(2), value);
  }
  return {
    file: args.get("file"),
    outfmt: args.get("outfmt"),
    width: Number(args.get("width") ?? 0),
    height: Number(args.get("height") ?? 0),
    page: Number(args.get("page") ?? 1),
    bgcolor: args.get("bgcolor") ?? "",
  };
}

function requireCredential(value, name) {
  if (!value) {
    throw new Error(`Missing ${name}. Set the environment variable or update payload headers.`);
  }
  return value;
}

async function toBase64(filePath) {
  const data = await fs.readFile(filePath);
  return data.toString("base64");
}

async function main() {
  const { file, outfmt, width, height, page, bgcolor } = parseArgs();
  if (!file || !outfmt) {
    throw new Error("Usage: node quickstart.mjs --file <path> --outfmt <format>");
  }

  const client = await Client.connect({
    url: DEFAULT_SERVER,
    headers: {
      ApiKey: requireCredential(API_KEY, "TWEAKIT_API_KEY"),
      ApiSecret: requireCredential(API_SECRET, "TWEAKIT_API_SECRET"),
    },
  });

  try {
    const tools = await client.listTools();
    console.log("Tools available:", tools.map((tool) => tool.name));

    const payload = {
      apiKey: API_KEY,
      apiSecret: API_SECRET,
      inext: path.extname(file).replace(".", "") || "bin",
      outfmt,
      blob: await toBase64(file),
      width,
      height,
      page,
      bgcolor,
    };

    const result = await client.callTool("convert", payload);
    if (result.isError) {
      console.error("Conversion failed:", result.error);
    } else {
      console.log("Conversion result:", result.data ?? result.structuredContent ?? result.content);
    }
  } finally {
    await client.close();
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: examples/python/quickstart.py
"""
Minimal Python quickstart for the TweekIT MCP server.

Run with:
    python examples/python/quickstart.py --file path/to/input.pdf --outfmt txt
"""

from __future__ import annotations

import argparse
import asyncio
from pathlib import Path

from clients.python.tweekit_client import TweekitClient


def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description="Convert a document using the TweekIT MCP server.")
    parser.add_argument("--file", type=Path, required=True, help="Path to the document to convert.")
    parser.add_argument("--outfmt", required=True, help="Desired output format (e.g. txt, pdf, png).")
    parser.add_argument("--width", type=int, default=0, help="Optional width for image resizing.")
    parser.add_argument("--height", type=int, default=0, help="Optional height for image resizing.")
    parser.add_argument("--page", type=int, default=1, help="Page to convert for multi-page documents.")
    parser.add_argument("--bgcolor", default="", help="Background color for transparent images (hex RGB).")
    return parser.parse_args()


async def main() -> None:
    args = parse_args()
    async with TweekitClient() as tweekit:
        tools = await tweekit.list_tools()
        print("Tools available:", tools)
        result = await tweekit.convert_file(
            path=args.file,
            outfmt=args.outfmt,
            width=args.width,
            height=args.height,
            page=args.page,
            bgcolor=args.bgcolor,
        )
        print("Conversion result:", result)


if __name__ == "__main__":
    asyncio.run(main())
